lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/evaluator/EvaluatorShim.java

```diff
@@ -277,9 +277,14 @@ private File downloadFile(final String url) throws IOException, StorageException
   private void extractFiles(final File zipFile) throws IOException {
     try (ZipFile zipFileHandle = new ZipFile(zipFile)) {
       Enumeration<? extends ZipEntry> zipEntries = zipFileHandle.entries();
+      Path reefPath = this.reefFileNames.getREEFFolder().toPath();
       while (zipEntries.hasMoreElements()) {
         ZipEntry zipEntry = zipEntries.nextElement();
-        File file = new File(this.reefFileNames.getREEFFolderName() + '/' + zipEntry.getName());
+        Path destination = new File(this.reefFileNames.getREEFFolder(), zipEntry.getName()).toPath();
```

Comment:
Please let me know if I am missing something here:
destination will always start with "reef" based on how you are constructing it. So the next line that checks if destination.startsWith(reefPath) will never fail. So the exception will never be thrown.

Reply:
I'm confused. I think what you have is definitely nicer code-wise, but functionality-wise, it looks like it would work the same. In the prior version, it too should start each entry with the reef folder name.
I know the JIRA references this code, but is this the correct issue? Have you been able to repro zip files being extracted outside the directory and confirmed this fixes it?